docs/reference/asyncapi-rules.md

```diff
@@ -0,0 +1,325 @@
+
+**Recommended:** Yes
+
+**Good Example**
+
+```yaml
+asyncapi: "2.0.0"
+info:
+  title: Awesome API
+  description: A very well defined API
+  version: "1.0"
+  contact:
+    name: A-Team
+    email: a-team@goarmy.com
+    url: https://goarmy.com/apis/support
+```
+
+### asyncapi2-info-contact
+
+Info object should contain `contact` object.
+
+Hopefully your API description document is so good that nobody ever needs to contact you with questions, but that is rarely the case. The contact object has a few different options for contact details.
+
+**Recommended:** Yes
+
+**Good Example**
+
+```yaml
+asyncapi: "2.0.0"
+info:
+  title: Awesome API
+  version: "1.0"
+  contact:
+    name: A-Team
+    email: a-team@goarmy.com
+```
+
+### asyncapi2-info-description
+
+OpenAPI object info `description` must be present and non-empty string.
+
+Examples can contain Markdown so you can really go to town with them, implementing getting started information like where to find authentication keys, and how to use them.
+
+**Recommended:** Yes
+
+**Good Example**
+
+```yaml
+asyncapi: "2.0.0"
+info:
+  version: "1.0.0"
+  title: Descriptive API
+  description: >+
+    Some description about the general point of this API, and why it exists when another similar but different API also exists.
+
+    ## Authentication
+
+    This API uses OAuth2 and tokens can be requested from [Dev Portal: Tokens](https://example.org/developers/tokens).
+```
+
+### asyncapi2-info-license-url
+
+Mentioning a license is only useful if people know what the license means, so add a link to the full text for those who need it.
+
+**Recommended:** No
+
+**Good Example**
+
+```yaml
+asyncapi: "2.0.0"
+info:
+  license:
+    name: MIT
+    url: https://www.tldrlegal.com/l/mit
+```
+
+### asyncapi2-info-license
+
+The `info` object should have a `license` key.
+
+It can be hard to pick a license, so if you don't have a lawyer around you can use [TLDRLegal](https://tldrlegal.com/) and [Choose a License](https://choosealicense.com/) to help give you an idea.
+
+How useful this is in court is not entirely known, but having a license is better than not having a license.
+
+**Recommended:** Yes
+
+**Good Example**
+
+```yaml
+asyncapi: "2.0.0"
+info:
+  license:
+    name: MIT
+```
+
+### asyncapi2-operation-description
+
+Operation objects should have a description.
+
+**Recommended:** Yes
+
+### asyncapi2-operation-operationId
+
+This operation ID is essentially a reference for the operation, which can be used to visually suggest a connection to other operations. This is like some theoretical static HATEOAS-style referencing, but it's also used for the URL in some documentation systems.
+
+Make the value `lower-hyphen-case`, and try and think of a name for the action which does not relate to the HTTP message. Base it off the actual action being performed. `create-polygon`? `search-by-polygon`? `filter-companies`?
```

Comment:
Why this recommendation? I'd say that camelCase is the one that works the best in most places. Most of the times the operationId is used for function or method names and URL hashes. `lower-hyphen-case` is usually not a friendly notation in programming languages and URL hashes also support camelCase.

Reply:
@fmvilas Dropped.